facades/PC/src/main/java/org/terasology/engine/Terasology.java

```diff
@@ -196,153 +219,121 @@ private static void setupLogging() {
+//        System.out.println("    terasology " + START_HEADLESS + " " + SERVER_PORT + TerasologyConstants.DEFAULT_PORT);
+//        System.out.println();
+//        System.out.println("    Load the latest game on startup and disable crash reporting");
+//        System.out.println("    terasology " + LOAD_LAST_GAME + " " + NO_CRASH_REPORT);
+//        System.out.println();
+//        System.out.println("    Don't start Terasology, just print this help:");
+//        System.out.println("    terasology " + PRINT_USAGE_FLAGS[1]);
+//        System.out.println();
+//        System.out.println("Alternatively use our standalone Launcher from: https://github.com/MovingBlocks/TerasologyLauncher/releases");
+//        System.out.println();
+//
         System.exit(0);
     }
 
-    private static void handleLaunchArguments(String[] args) {
-
-        Path homePath = null;
-
-        for (String arg : args) {
-            boolean recognized = true;
-
-            if (arg.startsWith(USE_SPECIFIED_DIR_AS_HOME)) {
-                homePath = Paths.get(arg.substring(USE_SPECIFIED_DIR_AS_HOME.length()));
-            } else if (arg.equals(USE_CURRENT_DIR_AS_HOME)) {
-                homePath = Paths.get("");
-            } else if (arg.equals(START_HEADLESS)) {
-                isHeadless = true;
-                crashReportEnabled = false;
-                splashEnabled = false;
-            } else if (arg.equals(NO_SAVE_GAMES)) {
-                System.setProperty(SystemConfig.SAVED_GAMES_ENABLED_PROPERTY, "false");
-            } else if (arg.equals(PERMISSIVE_SECURITY)) {
-                System.setProperty(SystemConfig.PERMISSIVE_SECURITY_ENABLED_PROPERTY, "true");
-            } else if (arg.equals(NO_CRASH_REPORT)) {
-                crashReportEnabled = false;
-            } else if (arg.equals(NO_SOUND)) {
-                soundEnabled = false;
-            } else if (arg.equals(NO_SPLASH)) {
-                splashEnabled = false;
-            } else if (arg.equals(LOAD_LAST_GAME)) {
-                loadLastGame = true;
-            } else if (arg.equals(CREATE_LAST_GAME)) {
-                createLastGame = true;
-            } else if (arg.startsWith(SERVER_PORT)) {
-                System.setProperty(ConfigurationSubsystem.SERVER_PORT_PROPERTY, arg.substring(SERVER_PORT.length()));
-            } else if (arg.startsWith(OVERRIDE_DEFAULT_CONFIG)) {
-                System.setProperty(Config.PROPERTY_OVERRIDE_DEFAULT_CONFIG, arg.substring(OVERRIDE_DEFAULT_CONFIG.length()));
-            } else {
-                recognized = false;
+    private void handleLaunchArguments() throws IOException {
+        if (homeDir != null) {
+            logger.info("homeDir is {}", homeDir);
+            if (homeDir.toString().startsWith("-")) {
+                // picocli shouldn't let this happen, but see
+                // https://github.com/remkop/picocli/issues/1197
```

Comment:
This has been fixed and is available in picocli v4.5.2 - do we still need this workaround?

Reply:
Correct, we can probably do without it now.

There are some edge cases involving what happens when you spell flags wrong, but going with the default behavior for that is a reasonable choice.

Reply:
this has been removed.